frontend/src/components/header/index.tsx

```diff
@@ -27,32 +27,28 @@ const ProfileLink = dynamic(
   { ssr: false },
 );
 
-const LanguagePicker: React.SFC<{ language: string }> = ({
-  language,
-  ...props
-}) => {
-  const alternateLinks = useAlternateLinks();
+const LanguagePicker = ({ language, ...props }: { language: string }) => {
+  const { asPath } = useRouter();
   return (
     <Flex sx={{ alignItems: "center", height: 50, mt: "-4px" }} {...props}>
-      <Link path={alternateLinks.en} sx={{ height: 40 }} external={true}>
+      <Link path={asPath} locale="en" sx={{ height: 40 }}>
         <EnglishIcon active={language === "en"} sx={{ width: 40, mr: 2 }} />
       </Link>
-      <Link path={alternateLinks.it} sx={{ height: 40 }} external={true}>
+      <Link path={asPath} locale="it" sx={{ height: 40 }}>
         <ItalianIcon active={language === "it"} sx={{ width: 40, mr: 4 }} />
       </Link>
     </Flex>
   );
 };
 
-const Links: React.SFC<{
+const Links = ({
+  links,
+}: {
   links: { href: string; title: string; page?: { slug: string } | null }[];
-}> = ({ links }) => (
+}) => (
   <Fragment>
     {links.map((link) => {
-      let path = link.page ? "/[lang]/[slug]" : link.href;
-
-      // nasty hack to make client side routing work with next.js
-      path = path.replace("/en", "/[lang]").replace("/it", "/[lang]");
+      const path = link.href;
```

Comment:
Will add (or remove?) page support better in another PR. I am writing it down and then write a ticket